### PR TITLE
Avoid resource collisions for CoreLoader protos

### PR DESCRIPTION
**Pull request by swankjesse**
We had bugs where we were non-deterministically loading these because multiple resources with the same path were on the classpath.

### CODE REVIEW

**Comment by swankjesse** [comment]
```
$ jar tf wire-schema/build/libs/wire-schema-jvm-5.0.0-SNAPSHOT.jar
...
CoreLoader-b51eb511d3/
CoreLoader-b51eb511d3/google/
CoreLoader-b51eb511d3/google/protobuf/
CoreLoader-b51eb511d3/google/protobuf/any.proto
CoreLoader-b51eb511d3/google/protobuf/descriptor.proto
CoreLoader-b51eb511d3/google/protobuf/duration.proto
CoreLoader-b51eb511d3/google/protobuf/empty.proto
CoreLoader-b51eb511d3/google/protobuf/struct.proto
CoreLoader-b51eb511d3/google/protobuf/timestamp.proto
CoreLoader-b51eb511d3/google/protobuf/wrappers.proto
CoreLoader-b51eb511d3/wire/
CoreLoader-b51eb511d3/wire/extensions.proto
```

**Comment by swankjesse** [comment]
```
$ cat wire-schema/build/coreLoader/kotlin/com/squareup/wire/schema/CoreLoaderResourceDirectory.kt
// Generated by CoreLoaderKotlinTask. Do not edit.
package com.squareup.wire.schema

internal const val coreLoaderResourceDirectory = "CoreLoader-b51eb511d3"
```

File: wire-schema/build.gradle.kts [hunk 1]
```diff
@@ -143,3 +151,108 @@ if (project.rootProject.name == "wire") {
...
+ * When we add Wire's `.proto` files like `google/descriptor.proto` as an element on the class path,
+ * we risk colliding with other `.jar` files that do the same, including other versions of Wire
+ * and with `protobuf-java.jar`.
+ *
+ * Mitigate this by putting Wire's copy of these protos in a directory named by the hash of its
+ * content.
+ *
+ * Also generate a single `.kt` file with this hash, so we can look it up later.
+ */
+fun Project.installCoreLoaderTasks() {
```
Comment (swankjesse) [inline wire-schema/build.gradle.kts:170]: I couldn’t move this to the build-support subproject because we need to build wire-schema before we build that!

**Review by oldergod — APPROVED** [review]
👍

**Comment by swankjesse** [comment]
Didn’t have the desired effect. I’m going to close this for now because it isn’t strictly required at the moment.

**Pull request closed without merging**